feat: update 96 - validator node registration (#143)
tari-project/tari-dot-com: _updates/2022-11-24-update-96.md

---
layout: update
tag: Developer Update
date: 2022-11-24
author: sdbondi 
thumbnail: update-background.jpg
title: Validator Node Registration
class: subpage
---

This week, draft [RFC-0313 - ValidatorNodeRegistration](https://rfc.tari.com/RFC-0313_VNRegistration.html) was merged

> and implemented for the igor network. This RFC describes the procedure for permissionless validator node registration 

Reviewer: ```suggestion
and implemented for the Igor network. This RFC describes the procedure for permissionless validator node registration 
```

and is the first step to participating in the Tari L2 network.

> This process is still under evaluation and some details may change as the development on the second layer continues.

Reviewer: ```suggestion
This procedure is under intense evaluation and many refinements are expected as development on the second layer continues.
```

confirmations before they are recognised to prevent inconsistencies that can occur from base chain reorgs. The base 
chain provides the validator node public key, the assigned shard key and a merkle root that commits to the current
validator set for each epoch.

Each validator node registration is valid for up to `VNRegistrationValidityPeriod` epochs and a validator may
re-register before the registration expires to ensure continued participation in L2 consensus. 

A number of new base layer consensus constants are added, namely

* `EpochLength` - defines the number of base chain blocks for an epoch
* `VNRegistrationValidityPeriod` - the validity period of a validator node registration UTXO
* `VNRegDepositAmount` - the minimum amount of base layer coins that must be locked up in the validator node registration UTXO
* `VNRegLockHeight` - the number of epochs that funds in the validator node registration UTXO must be locked
* `VNShardShuffleInterval` - the number of epochs before the validator node shard key will be pseudorandomly reassigned

Additionally, a new block header field `validator_node_mr` is added that commits the validator set for the current epoch.

> At the time of writing, there are no igor seed nodes or a validator node network, so you might want to wait for these before trying this out.

Reviewer: ```suggestion
At the time of writing, there are no Igor seed nodes or a validator node network, so you might want to wait for these before trying this out.
```
